### DISCUSSION TITLE
Simple way to set cellRenderer in agGrid

### DISCUSSION THREAD

**Discussion by aersam** [zauberzeug/nicegui#1265] [Ideas / Feature Requests]
Hi guys!

I want to set a cellRenderer in ui.aggrid as a JavaScript Function which seems to be pretty complicated. I now do something like this:

```python
   links_renderer = """params => `<span style="color: ${params.color}">${params.value}</span>`"""
    await ui.run_javascript(f'''
        getElement({grid.id}).gridOptions.columnApi.getColumn("id").getColDef().cellRenderer = {links_renderer};
        getElement({grid.id}).gridOptions.api.refreshCells();
    ''', respond=False)
```

But it's pretty hard to get the timing right. In an ideal world I could do something like this:

```python
from nicegui import JavaScriptFunction 
grid = ui.aggrid({
    'columnDefs': [
        {'headerName': 'id', "field": "id", "cellRendererParams": {"color": "green"}, "cellRenderer": JavaScriptFunction("""params => `<span style="color: ${params.color}">${params.value}</span>`""")},
        {'headerName': 'Name', 'field': 'name'},
    ],
    'rowData': []
})

```

**Comment by falkoschindler** [comment]
I agree! Passing JavaScript functions via `columnDefs` would be powerful.

Just as a side note: Do you know the [`html_columns` parameter](https://nicegui.io/documentation/aggrid#render_columns_as_html)? Could be useful in your case.

**Comment by aersam** [comment]
Yes html_columns would work as a work around for now. But will require me to generate lot's of html which I actually want to avoid if possible
